Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs

```diff
@@ -0,0 +1,311 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using System.Management.Automation;
+using System.Management.Automation.Language;
+using System.Management.Automation.Runspaces;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Interactive.Commands;
+using Microsoft.DotNet.Interactive.Events;
+using Microsoft.DotNet.Interactive.Formatting;
+
+namespace Microsoft.DotNet.Interactive.PowerShell
+{
+    public class PowerShellKernel : KernelBase, IExtensibleKernel
+    {
+        internal const string DefaultKernelName = "powershell";
+
+        private Runspace _runspace;
+        private SemaphoreSlim _runspaceSemaphore;
+        private CancellationTokenSource _cancellationSource;
+        private readonly object _cancellationSourceLock = new object();
+
+        public PowerShellKernel()
+        {
+            var iss = InitialSessionState.CreateDefault();
+            _runspace = RunspaceFactory.CreateRunspace(iss);
+            _runspace.Open();
+            _runspaceSemaphore = new SemaphoreSlim(1, 1);
+            _cancellationSource = new CancellationTokenSource();
+            Name = DefaultKernelName;
+
+            string psModulePath = Environment.GetEnvironmentVariable("PSModulePath");
+            Environment.SetEnvironmentVariable("PSModulePath",
+                psModulePath + Path.PathSeparator +
+                System.IO.Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location) + "/Modules");
```

Comment:
You probably want to remove "/" from "/Modules".

Using `Path.Combine` will let you avoid having to specify the path separators.